active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,230 @@
+
+### Extensible Resource
+
+An Azure data-plane resource or non-Azure resource declared in a Bicep file or an ARM template. Extensible resources are deployed through extensibility providers.
+
+## Motivation
+
+Users currently encounter a major challenge when deploying extensible resources, as there is inadequate information available when querying the deployments or deployment operation APIs. Key details, such as resource IDs, are not applicable to extensible resources, making it difficult for users to monitor the progress of their deployments. For instance, Azure Portal users attempting to view extensible resource deployment on the deployment details blade will only see an infinite loading animation without any details being displayed. Similarly, Azure PowerShell users tracking deployment progress using the deployment cmdlet and the `-Verbose` switch will receive logs like `VERBOSE: 00:00:00 - Resource '' provisioning status is succeeded`.
+
+To address this issue, the deployments and deployment operations APIs must be updated to include additional information, such as:
+
+- Resource symbolic name
+- Resource type
+- Deployment provider alias
+- Deployment provider name
+- Deployment provider version
+
+This additional data would enable users to identify a deployed extensible resource and map it to the corresponding resource definition in a Bicep file or an ARM template, providing users with comprehensive insights.
+
+## Detailed design
+
+> The core of the REP. Elaborate on the design and implementation with sufficient detail for someone familiar with Bicep to grasp. Provide detailed examples to illustrate how the feature is used and its implications on user experience.
+
+### `Microsoft.Resources/deployments` API changes for PUT and GET
+
+#### Adding `deploymentProviders`
+
+The `Microsoft.Resources/deployments` API will be updated to include a `deploymentProviders` array in both the PUT and GET response bodies. This modification is intended to incorporate information about deployment providers that are involved in executing the deployment to offer a more comprehensive deployment view.
+
+> It's worth noting that we cannot use the name `providers` as the property `providers` is already assigned to return Azure Resource Provider information. While an alternative approach could involve renaming the current `providers` property to `resourceProviders` for enhanced accuracy, the solution should be avoided as it constitutes a breaking change.
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    ...
+    "providers": [
+      {
+        "namespace": "Microsoft.Storage",
+        "resourceTypes": [
+          {
+            "resourceType": "storageAccounts",
+            "locations": [
+              "eastus"
+            ]
+          }
+        ]
+      }
+    ],
++   "deploymentProviders": [
```

Comment:
"deployment" & "providers" are both pretty heavily overloaded and commonly used. What about something like "importedProviders"?

Reply:
Makes sense. Added a comment to mention this as an alternative, and added it as an unresolved question. Personally, I don't have a strong opinion on the property name, and I think we can do a vote during next week's Bicep discussion.

Sharing some other alternatives suggested by ChatGPT:
- infraProviders
- infrastructureProviders
- platformProviders
- platforms